Add ability to pass extensions to `make-json` command

michaelw85 opened the pull request:
Resolves #231

swissspidy commented:
Thanks for opening this PR, appreciate the initiative!

Please note that most tests here are Behat tests instead of PHPUnit tests, see `features/makejson.feature`. We should follow that in this PR.

michaelw85 commented:
> Thanks for opening this PR, appreciate the initiative!
> 
> Please note that most tests here are Behat tests instead of PHPUnit tests, see `features/makejson.feature`. We should follow that in this PR.

Alright, I thought it would be pretty easy to unit test (did struggle a bit with mocking though 😅 ) I will have a look at the Behat tests and add some.

michaelw85 commented:
@swissspidy I've added some basic scenarios similar to the unit tests and everything passes 🥳

swissspidy commented on tests/MakeJsonTest.php:
```diff
@@ -0,0 +1,101 @@
+	public function set_up() {
+		parent::set_up();
+
+		/**
+		 * PHP5.4 cannot set property with __DIR__ constant.
+		 * Shamelessly stolen from @see IterableCodeExtractorTest.php
+		 */
+		self::$base = Utils\normalize_path( __DIR__ ) . '/data/make_json/';
```
We don't support PHP 5.4 (anymore)

```suggestion
		self::$base = Utils\normalize_path( __DIR__ ) . '/data/make_json/';
```

michaelw85 commented on foo/empty.mo:
These files are still used in the unit tests

swissspidy replied:
Then these files should be moved to the `tests/data` folder like the others.

swissspidy replied:
Alternatively we can also just keep the Behat test and remove the unit test. Less need for creating these dummy files.

michaelw85 replied:
I did place these files in the data dir, weird. Guess these were generated by accident. I will check what happened and see if I will remove the unit tests

michaelw85 replied:
Removed file

michaelw85 commented on src/MakeJsonCommand.php:
```diff
@@ -251,18 +262,12 @@ protected function make_json( $source_file, $destination, $map, $domain ) {
-					}
-
-					return null;
+					return in_array( $extension, $extensions, true ) ? str_replace( '.min.', '.', $file ) : null;
```
Yes you are right that's why I added two dots to make this less likely but I will update and improve this code

michaelw85 replied:
Improved check, specifically replacing .min.{ext} at the end of the string now.

michaelw85 commented on foo/translations.mo:
Removed file

michaelw85 commented:
@swissspidy I've updated the PR, resolved your feedback. Removed files and the new unit tests in favor of Behat tests, and lastly improved removing `.min` from minified files. Let me know if you see anything else that needs improvement.
